x/dsmr/storage.go

```diff
@@ -141,14 +202,20 @@ func (s *ChunkStorage[T]) AddLocalChunkWithCert(c Chunk[T], cert *ChunkCertifica
 }
 
 // SetChunkCert sets the chunk certificate for the given chunkID
-func (s *ChunkStorage[T]) SetChunkCert(chunkID ids.ID, cert *ChunkCertificate) error {
+func (s *ChunkStorage[T]) SetChunkCert(ctx context.Context, chunkID ids.ID, cert *ChunkCertificate) error {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
 	storedChunk, ok := s.chunkMap[chunkID]
 	if !ok {
 		return fmt.Errorf("failed to store cert for non-existent chunk: %s", chunkID)
 	}
+	if cert == nil {
+		return ErrInvalidChunkCertificate
+	}
```

Comment:
If we are changing this where the cert cannot be nil, could we remove this check and make it an assumption that this cannot be called with a nil chunk certificate?